compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -192,29 +194,32 @@
  *
  * @since 2201.3.0
  */
-public class ParameterDesugar extends BLangNodeVisitor {
-    private static final CompilerContext.Key<ParameterDesugar> PARAMETER_DESUGAR_KEY = new CompilerContext.Key<>();
+public class ClosureGenerator extends BLangNodeVisitor {
```

Comment:
I believe I raised this previously also. Why did we implement this as a node visitor instead of a node analyzer?

Reply:
Create a separate issue to address this(https://github.com/ballerina-platform/ballerina-lang/issues/39334)